### ISSUE TITLE
Internstatus og internkort

### ISSUE THREAD

**Issue by henrist** [cybrairai/internsystem#14]
Internsystemet bør utvides til å ha støtte for å registrere interne, samt generere internkort. Se også #8.

I dag gjøres dette på wikien, se f.eks. https://confluence.cyb.no/display/AKTIV/Internkort+V15.

For hvert semester lages det nye lister.

Til forskjell fra løsningen på wiki bør det være mulig å ha flere roller, f.eks. både webfunk og barfunk. De ulike rollene bør derfor lagres som en separet modell med en én-til-mange-relasjon fra internkort-lista.

# Modellering
Modell `intern`:
* Personen det gjelder (referanse til bruker-objekt, bør være enkelt å opprette nye)
* Utgiver (referanse til bruker-objekt)
* Hvorvidt internkort er generert
* Hvorvidt personen er aktiv (for tilbaketrekking av internstatus)
* Kommentarfelt

Modell `intern_roles`
* Navn på rolle
* Navn på gruppe denne rollen tilhører (f.eks. "Hovedstyret") - kan utvides til egen relasjon senere

Mange-til-mange-relasjon mellom `intern` og `intern_roles`.

# Rettigheter
* Må avklares. Kan foreløpig begrense dette til admins og utvide senere.

# Generering av internkort
Det ligger et script på cyb-brukeren for generering av internkort. Må være obs så ikke dette lekker ut med designet, slik at alle kan opprette internkort.

# Annet
Det bør være mulig å se hvem av de interne som ikke har kjøpt medlemsskap, evt. sperres for generering av internkort hvis man ikke har medlemsstatus.

**Comment by henrist** [comment]
Oppdatert en del info nå. Blant annet innført `intern_group` etter innspillene som kom på #8.

**Comment by matsasc** [comment]
Helt enig at koden som generer internkort og medlemskort ikke burde ligge offentlig tilgjenglig. En idé kan jo være å skrive en liten service som genererer internkort, medlemskort og livstidsmedlemskapskort. Evt så kan man bare lage ut lister med riktig formattering for scriptet og kjøre de manuelt senere(det scriptet burde refaktoreres uansett, er ganske rotete)

Utenom det så ser modellene veldig fine ut. Eneste jeg savner med det er hvilke korttilganger rollen burde ha.

**Comment by henrist** [comment]
God idé å skille ut genereringen som en egen service, det kan jo funke ganske greit. I starten er det kanskje greiest med idéen om å eksportere en oversikt som lese av disse scriptene.

Når du snakker om korttilganger, så tenker du vel på den rollen systemet til Vegard har? I så fall er jeg helt enig i at det kunne vært flott og hatt integrert i dette systemet på sikt. Men det bør være forholdsvis enkelt å utvide til det senere.

**Comment by matsasc** [comment]
Ja, personlig mener jeg at man også burde få det inn mens man designer det fra starten av. Gjør ikke oppgaven veldig mye mer komplisert(man trenger nok 1 eller 2 ekstra modeller og litt tenking) og man slipper enda en login man må huske. Jeg kan komme med noen forslag til modeller i morgen.

**Comment by henrist** [comment]
Vel, mye arbeid på frontend som må på plass som kompliserer en god del. Og når vi har en fungerende løsning i dag, så kan det være bedre å spare på ressursene. Dog enig i at det er herk å ha pålogging flere steder.

**Comment by henrist** [comment]
Bør det være mulig å ha en kommentar, evt. en slags "tag" på mange-til-mange-relasjonen mellom `intern_person` og `intern_role`? F.eks. at man for arrmester kan skrive en mer forklarende tekst? Si f.eks. at det kan stå Arrmester (Ifi-skitur 2016), hvor da "Ifi-skitur 2016" er teksten man legger inn?
